Makefile

```diff
@@ -19,7 +19,8 @@ MLRUN_VERSION ?= unstable
 # version for the python package with 0.0.0+
 # if the provided version includes a "+" we replace it with "-" for the docker tag
 MLRUN_DOCKER_TAG ?= $(shell echo "$(MLRUN_VERSION)" | sed -E 's/\+/\-/g')
-MLRUN_PYTHON_PACKAGE_VERSION ?= $(MLRUN_VERSION)
+# if the provided version includes a "-" we replace its first occurrence with "+" to align with PEP 404
+MLRUN_PYTHON_PACKAGE_VERSION ?= $(shell echo "$(MLRUN_VERSION)" | sed 's/\-/\+/')
```

Comment:
That's not good, if the provided version is `0.1.2+some-thing` it is totally valid and with this code you'll change it to `0.1.2+some+thing`
You want to do this change only when the input version looks like `<number>.<number>.<number>-`